config/tools/config_generator.cpp

```diff
@@ -0,0 +1,896 @@
+                return as_int;
+            }
+            const auto as_dbl = std::stod(value);
+            return as_dbl;
+        }
+
+        if(keep_quotes == false) {
+            const auto unquoted = value.substr(1, value.size() - 2);
+            return unquoted;
+        }
+        return value;
+    }
+
+    [[nodiscard]] auto config_generator::get_param_from_template_file(
+        const std::string option,
+        std::map<std::string, std::string>& config_map)
+        -> std::variant<std::string, size_t, double> {
+        auto it = config_map.find(option);
+        if(it != config_map.end()) {
+            value_t parsed_val = parse_value(it->second, false);
+            if(std::holds_alternative<size_t>(parsed_val)) {
+                return std::get<size_t>(parsed_val);
+            } else if(std::holds_alternative<double>(parsed_val)) {
+                return std::get<double>(parsed_val);
+            } else if(std::holds_alternative<std::string>(parsed_val)) {
+                return std::get<std::string>(parsed_val);
+            } else {
+                __builtin_unreachable();
+            }
+        } else {
+            std::string error_msg
+                = "Warning: Could not find param, " + option + ".";
+            std::cout << error_msg << std::endl;
+            return error_msg;
+        }
+    }
+
+    void config_generator::set_param_to_config_file(const std::string key,
+                                                    const std::string value) {
+        m_new_config << key << "=" << '"' << value << '"' << '\n';
+    }
+
+    void config_generator::set_param_to_config_file(const std::string key,
+                                                    const double value) {
+        m_new_config << key << "=" << value << '\n';
+    }
+
+    void config_generator::set_param_to_config_file(const std::string key,
+                                                    const size_t value) {
+        m_new_config << key << "=" << value << '\n';
+    }
+
+    void config_generator::set_log_level(const std::string key,
+                                         std::string& log_level) {
+        if(log_levels.find(log_level) == log_levels.end()) {
+            log_level = "DEBUG";
+            std::cout << "Warning: Log level not recognized. Setting to DEBUG"
+                      << std::endl;
+        } else if(find_value<std::string>(key).has_value() == false) {
+            log_level
+                = find_value<std::string>(tmpl_universal_override_log_level)
+                      .value();
+        }
+    }
+
+    [[maybe_unused]] auto
+    config_generator::create_component(const char* type,
+                                       const value_t component_count,
+                                       bool create_2pc) -> std::string {
+        std::string return_msg = "";
+        if(std::holds_alternative<size_t>(component_count)) {
```

Comment:
I think the point of this is solely to ensure that `component_count` is a `size_t`. If it's not, it returns an empty string as a message. Given that, I'd suggest changing the `component_count` argument to a `size_t` and doing `std::get<>` at the call-site.

Reply:
DONE

Reply:
Sorry, to clarify, I meant that the signature of the function should be changed to have `component_count` be a `size_t`. At that point, it's impossible to call the function with an empty string as the `component_count`, and it's the call-site's job to decide what to pass and to handle the error locally if the thing it wants to use is not valid.

Reply:
Ok. I did change the function signature to take that as size_t